demos/speech_recognition_demo/python/utils/audio_features.py

```diff
@@ -1,9 +1,79 @@
 #
-# Copyright (C) 2020 Intel Corporation
+# Copyright (C) 2020-2021 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 #
+from copy import deepcopy
+
 import numpy as np
-import librosa
+
+# Workaround to import librosa on Linux without installed libsndfile.so
+try:
+    import librosa
+except OSError:
+    import sys
+    import types
+    sys.modules['soundfile'] = types.ModuleType('fake_soundfile')
+    import librosa
+    del sys.modules['soundfile']
```

Comment:
```suggestion
```
https://github.com/openvinotoolkit/open_model_zoo/pull/2054#discussion_r622920988

Reply:
I checked that deleting `sys.modules['soundfile']` affects importing `soundfile` in the other modules (by adding `raise OSError()` into `try:` and adding `import soundfile; print(dir(soundfile))` into `speech_recognition_demo.py`). With deletion the correct `soundfile` is imported, and without deletion the fake `soundfile` is imported.

`soundfile` may be used for loading audio in the user's code or inside the packages he imports. When importing it people sometimes check for errors to give a comprehensible error message. If we didn't clean up, we'd break this. An example of such checking inside `fairseq`: https://github.com/pytorch/fairseq/blob/425c36eafff535fe7337f8bdd5ace22ebacc78cb/fairseq/data/audio/audio_utils.py#L57 . And even without checking, instead of `ImportError` in proper place we'll have some other incomprehensible error in another place.

`librosa` is actually left in a broken state this way. I think it's more-or-less OK, because we're leaving only one package in a partially broken state, as opposed to leaving completely broken `soundfile`.  If needed, `librosa` can be cleaned up by deleting three cached modules: `librosa`, `librosa.core` and `librosa.core.audio` -- I thought that it may look too complicated for its goal and may be brittle to changes inside `librosa`.

Reply:
> With deletion the correct `soundfile` is imported, and without deletion the fake `soundfile` is imported.

That's what we want. If demo import can't find libsndfile it is reasonble to assume that other similar imports also won't find it. You never know if such imports are going to be used for file reading or a case similar to ours.

> soundfile may be used for loading audio in the user's code or inside the packages he imports.

The solution is a hack anyway. For example I could say that one could copypaste the demo code with that import and than try reading file somhere in `audio_features.py`. The problem is it was never defined how they are suppoed to reuse the code because OMZ demos implementation is not expected to be used blindly. It's not up to us to change the whole OMZ position in this PR.

I'm not much into this discussion, I would prefer to leave it up to you and @IRDonch :)

Reply:
Just realized, if Roman ignores this, it is not defined what to do next. You can go with your solution (and update speech_recognition_offline_demo the same way) unless there are complains.

Reply:
I don't think we need to worry about what users will do in their own code. This hack is clearly not something that should be used in production code, whether with the deletion or without. We're only doing it so that we can spare the users the trouble of installing `libsndfile` on their computer just to run the demo.

Also FWIW, the proper solution to this would be to open an issue on the librosa tracker and ask the maintainers nicely to make the soundfile dependency optional (i.e. put it in an extra).

Reply:
OK then, removed the clean-up.